vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_event_util_unittest.cc

```diff
@@ -0,0 +1,190 @@
+  ad_events.push_back(ad_event_4);
+
+  const AdEventInfo ad_event_3 =
+      GetAdEvent(creative_ad_notification_1, ConfirmationType::kViewed,
+                 now - base::TimeDelta::FromHours(6));
+  ad_events.push_back(ad_event_3);
+
+  const AdEventInfo ad_event_2 =
+      GetAdEvent(creative_ad_notification_2, ConfirmationType::kViewed,
+                 now - base::TimeDelta::FromHours(11));
+  ad_events.push_back(ad_event_2);
+
+  const AdEventInfo ad_event_1 =
+      GetAdEvent(creative_ad_notification_1, ConfirmationType::kViewed,
+                 now - base::TimeDelta::FromHours(12));
+  ad_events.push_back(ad_event_1);
+
+  // Act
+  const absl::optional<base::Time> ad_last_seen =
+      GetLastSeenAdTime(ad_events, creative_ad_notification_1);
+
+  // Assert
+  const base::Time expected_timestamp = now - base::TimeDelta::FromHours(6);
+  const base::Time expected_ad_last_seen = base::Time::FromDoubleT(
+      static_cast<int64_t>(expected_timestamp.ToDoubleT()));
+  EXPECT_EQ(expected_ad_last_seen, ad_last_seen.value());
+}
+
+TEST(BatAdsAdEventUtilTest, GetLastSeenAdvertiserTimeForEmptyAdEvents) {
+  // Arrange
+  AdEventList ad_events;
+
+  const CreativeAdNotificationInfo creative_ad_notification =
+      GetCreativeAdNotification();
+
+  // Act
+  const absl::optional<base::Time> advertiser_last_seen =
+      GetLastSeenAdvertiserTime(ad_events, creative_ad_notification);
+
+  // Assert
+  EXPECT_EQ(absl::nullopt, advertiser_last_seen);
+}
+
+TEST(BatAdsAdEventUtilTest, GetLastSeenAdvertiserTimeForUnseenAdvertiser) {
+  // Arrange
+  AdEventList ad_events;
+
+  const CreativeAdNotificationInfo creative_ad_notification_1 =
+      GetCreativeAdNotification();
+
+  const base::Time event_time =
+      base::Time::Now() - base::TimeDelta::FromHours(12);
+  const AdEventInfo ad_event = GetAdEvent(
+      creative_ad_notification_1, ConfirmationType::kViewed, event_time);
+  ad_events.push_back(ad_event);
+
+  // Act
+  const CreativeAdNotificationInfo creative_ad_notification_2 =
+      GetCreativeAdNotification();
+  const absl::optional<base::Time> advertiser_last_seen =
+      GetLastSeenAdvertiserTime(ad_events, creative_ad_notification_2);
+
+  // Assert
+  EXPECT_EQ(absl::nullopt, advertiser_last_seen);
+}
+
+TEST(BatAdsAdEventUtilTest, GetLastSeenAdvertiserTime) {
+  // Arrange
+  AdEventList ad_events;
+
+  const base::Time now = base::Time::Now();
+
+  const std::string advertiser_id_1 = base::GenerateGUID();
+  const CreativeAdNotificationInfo creative_ad_notification_1 =
+      GetCreativeAdNotification(advertiser_id_1);
+
+  const std::string advertiser_id_2 = base::GenerateGUID();
+  const CreativeAdNotificationInfo creative_ad_notification_2 =
+      GetCreativeAdNotification(advertiser_id_2);
+
+  const CreativeAdNotificationInfo creative_ad_notification_3 =
+      GetCreativeAdNotification(advertiser_id_1);
+
+  // Ad events are ordered by date DESC in |AdEvents::GetAll()|
```

Comment:
nit: maybe rename GetAll to pass a sort type?

Reply:
I'd say not for now, but probably something to consider for other db calls as well to make ordering explicit